### PR TITLE
Popcorn clean

### PR DESCRIPTION
**Pull request by Morriar**
Just some cleaning in popcorn modules

Also introducing mongo options for pop_repos.

### CODE REVIEW

File: lib/popcorn/pop_auth.nit [hunk 1]
```diff
@@ -80,6 +80,15 @@ module pop_auth
 import pop_handlers
 import github
 
+# Initialize session in request if non existent.
+#
+# Should be called before any use of the session.
+class SessionInit
+	super Handler
+
+	redef fun all(req, res) do if req.session == null then req.session = new Session
```
Comment (privat) [inline lib/popcorn/pop_auth.nit]: `pop_auth` requites the importation of github, isn't that too much?
One might assume that `SessionInit` is light and useful enough to be provided by default

**Comment by privat** [comment]
+1

**Comment by Morriar** [comment]
Reroll: I moved SessionInit to its own module `pop_sessions`.

**Comment by privat** [comment]
http://gresil.org/jenkins/view/Int%C3%A9gration/job/nit_test_tools/1088/testReport/junit/nitunit.popcorn/AuthHandler/_class_/

**Comment by Morriar** [comment]
test this please

**Pull request merged**